<?xml version="1.0" ?>
<robot name="three_DOF_robot">
	
	<!-- Used for fixing robot to Gazebo 'base_link' -->
	<link name="world" />
	<joint name="glue_robot_to_world" type="fixed">
		<parent link="world" />
		<child link="link1" />
	</joint>
	<!-- Base Link -->
	<link name="link1">
		<collision>
			<origin xyz="0 0 0.5" rpy="0 0 0" />
			<geometry>
				<box size="0.2 0.2 0.7" />
			</geometry>
		</collision>
		<visual>
			<origin xyz="0 0 0.5" rpy="0 0 0" />
			<geometry>
				<box size="0.2 0.2 1" />
			</geometry>
		</visual>
		<inertial>
			<origin xyz="0 0 0.5" rpy="0 0 0" />
			<mass value="0.5" />
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
		</inertial>
	</link>
	<!-- Moveable Link -->
	<link name="link2">
		<collision>
			<origin xyz="0 0 0.5" rpy="0 0 0" />
			<geometry>
				<cylinder length="1" radius="0.1" />
				<!--box size="0.15 0.15 0.8"-->
			</geometry>
		</collision>
		<visual>
			<origin xyz="0 0 0.5" rpy="0 0 0" />
			<geometry>
				<cylinder length="1" radius="0.1" />
			</geometry>
		</visual>
		<inertial>
			<origin xyz="0 0 0.5" rpy="0 0 0" />
			<mass value="0.3" />
			<inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.005"/>
		</inertial>
	</link>
	
	<joint name="joint1" type="continuous">
		<parent link="link1" />
		<child link="link2" />
		<origin xyz="0 0 1" rpy="0 0 0" />
		<axis xyz="0 1 0" />
	</joint>
	
	<link name="connector">
		<visual>
			<origin xyz="0 0 0.075" rpy="0 0 0" />
			<geometry>
				<sphere radius="0.15" />
			</geometry>
			<material name="black">
				<color rgba="0 0 0 1"/>
			</material>
		</visual>
		<collision>
			<origin xyz="0 0 0.075" rpy="0 0 0" />
			<geometry>
				<sphere radius="0.1" />
			</geometry>
		</collision>
		<inertial>
			<origin xyz="0 0 0.075" rpy="0 0 0" />
			<mass value="0.05" />
			<inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.005"/>
		</inertial>
	</link>
	<link name="link3">
		<collision>
			<origin xyz="0 0 0.4" rpy="0 0 0" />
			<geometry>
				<cylinder length="0.8" radius="0.1" />
			</geometry>
		</collision>
		<visual>
			<origin xyz="0 0 0.4" rpy="0 0 0" /> 
			<geometry>
				<cylinder length="0.8" radius="0.1" />
			</geometry>
			<material name="blue">
				<color rgba="0 0 .8 1"/>
			</material>
		</visual>
		<inertial>
		 	<origin xyz="0 0 0.4" rpy="0 0 0" /> 
			<mass value="0.1" />
			<inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.005"/>
		</inertial>
	</link>
	<joint name="joint2" type="continuous">
		<parent link="link2" />
		<child link="connector" />
		<origin xyz="0 0 1.025" rpy="0 0 0" />
		<axis xyz="0 0 1" />
	</joint>
	<joint name="joint3" type="continuous">
		<parent link="connector" />
		<child link="link3" />
		<origin xyz="0 0 0.075" rpy="0 0 0" />
		<axis xyz="0 1 0" />
	</joint>
</robot>
